src/otx/v2/adapters/torch/mmengine/mmdet/dataset.py

```diff
@@ -0,0 +1,180 @@
+
+        Args:
+            task (Optional[Union[TaskType, str]], optional): The task type of the dataset want to load.
+                Defaults to None.
+            train_type (Optional[Union[TrainType, str]], optional): The train type of the dataset want to load.
+                Defaults to None.
+            train_data_roots (Optional[str], optional): The root address of the dataset to be used for training.
+                Defaults to None.
+            train_ann_files (Optional[str], optional): Location of the annotation file for the dataset
+                to be used for training. Defaults to None.
+            val_data_roots (Optional[str], optional): The root address of the dataset
+                to be used for validation. Defaults to None.
+            val_ann_files (Optional[str], optional): Location of the annotation file for the dataset
+                to be used for validation. Defaults to None.
+            test_data_roots (Optional[str], optional): The root address of the dataset
+                to be used for testing. Defaults to None.
+            test_ann_files (Optional[str], optional): Location of the annotation file for the dataset
+                to be used for testing. Defaults to None.
+            unlabeled_data_roots (Optional[str], optional): The root address of the unlabeled dataset
+                to be used for training. Defaults to None.
+            unlabeled_file_list (Optional[str], optional): The file where the list of unlabeled images is declared.
+                Defaults to None.
+            data_format (Optional[str], optional): The format of the dataset. Defaults to None.
+        """
+        super().__init__(
+            task,
+            train_type,
+            train_data_roots,
+            train_ann_files,
+            val_data_roots,
+            val_ann_files,
+            test_data_roots,
+            test_ann_files,
+            unlabeled_data_roots,
+            unlabeled_file_list,
+            data_format,
+        )
+        self.scope = "mmdet"
+        self.dataset_registry = MMDetRegistry().get("dataset")
+
+    def _get_sub_task_dataset(self) -> TorchDataset:
+        return OTXDetDataset
+
+    def _build_dataset(
+        self,
+        subset: str,
+        pipeline: list | None = None,
+        config: dict | None = None,
+    ) -> TorchDataset | None:
+        """Builds a TorchDataset object for the given subset using the specified pipeline and configuration.
+
+        Args:
+            subset (str): The subset to build the dataset for.
+            pipeline (Optional[Union[list, dict]]): The pipeline to use for the dataset.
+                Defaults to None.
+            config (Optional[Union[str, dict]]): The configuration to use for the dataset.
+                Defaults to None.
+
+        Returns:
+            Optional[TorchDataset]: The built TorchDataset object, or None if the dataset is empty.
+        """
+        if pipeline is None:
```

Comment:
don't we need to check whether dataset configurations are existed in config or not?
Please refer to https://github.com/openvinotoolkit/training_extensions/blob/e6c7325c506e93dec7c1b6a5ab9e63bc80a9d50c/src/otx/v2/adapters/torch/mmengine/mmpretrain/dataset.py#L144

Reply:
Thanks for detailed review, it is reflected by 
f677a96